### ISSUE TITLE
Parameter matching for unpacked TypedDict not enforced

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#9808]
The typing spec indicates that if a caller passes an unpacked (non-closed) TypedDict as an argument to a function that accepts a fixed set of keyword argument, it should result in an error because additional items may be present in the argument TypedDict.

```
# > Kwargs hinted with an unpacked TypedDict can only be passed to another function if the function
# > to which unpacked kwargs are being passed to has **kwargs in its signature as well,
# > because then additional keywords would not cause errors at runtime during function invocation. 
# > Otherwise, the type checker should generate an error.
```

For more details, refer to [this pull request](https://github.com/python/typing/pull/1918) in the conformance tests.

**Comment by erictraut** [comment]
After thinking about this more, I agree that the typing spec is inconsistent in this case and should be changed for consistency. I'm therefore not going to make this change. Refer to the above PR link for further discussion and resolution.
